pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -40,34 +41,62 @@ func retainPodFields(desired, observed *unstructured.Unstructured) (*unstructure
+	// Pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`,
+	// `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or
+	// `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative).
+	// More info: https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement
+	clusterPod.Labels = desiredPod.Labels
+	clusterPod.Annotations = desiredPod.Annotations
+	updatePodImages(desiredPod, clusterPod)
+	updatePodActiveDeadlineSeconds(desiredPod, clusterPod)
+	updatePodTolerations(desiredPod, clusterPod)
+
+	unstructuredObj, err := helper.ToUnstructured(clusterPod)
+	if err != nil {
+		return nil, fmt.Errorf("failed to transform Pod: %v", err)
+	}
+
+	return unstructuredObj, nil
+}
+
+func updatePodImages(desiredPod, clusterPod *corev1.Pod) {
+	for clusterIndex, clusterContainer := range clusterPod.Spec.InitContainers {
+		for _, desiredContainer := range desiredPod.Spec.InitContainers {
 			if desiredContainer.Name == clusterContainer.Name {
-				desiredPod.Spec.Containers[desiredIndex].VolumeMounts = clusterContainer.VolumeMounts
+				clusterPod.Spec.InitContainers[clusterIndex].Image = desiredContainer.Image
 				break
 			}
 		}
 	}
-
-	// retain volumeMounts in each init container
-	for _, clusterInitContainer := range clusterPod.Spec.InitContainers {
-		for desiredIndex, desiredInitContainer := range desiredPod.Spec.InitContainers {
-			if desiredInitContainer.Name == clusterInitContainer.Name {
-				desiredPod.Spec.InitContainers[desiredIndex].VolumeMounts = clusterInitContainer.VolumeMounts
+	for clusterIndex, clusterContainer := range clusterPod.Spec.Containers {
+		for _, desiredContainer := range desiredPod.Spec.Containers {
+			if desiredContainer.Name == clusterContainer.Name {
+				clusterPod.Spec.Containers[clusterIndex].Image = desiredContainer.Image
 				break
 			}
 		}
 	}
-	unstructuredObj, err := helper.ToUnstructured(desiredPod)
-	if err != nil {
-		return nil, fmt.Errorf("failed to transform Pod: %v", err)
+}
+
+func updatePodActiveDeadlineSeconds(desiredPod, clusterPod *corev1.Pod) {
+	desiredActiveDeadlineSeconds := pointer.Int64Deref(desiredPod.Spec.ActiveDeadlineSeconds, 0)
+	clusterActiveDeadlineSeconds := pointer.Int64Deref(clusterPod.Spec.ActiveDeadlineSeconds, 0)
+	if (clusterPod.Spec.ActiveDeadlineSeconds == nil && desiredActiveDeadlineSeconds > 0) ||
+		(desiredActiveDeadlineSeconds > 0 && desiredActiveDeadlineSeconds < clusterActiveDeadlineSeconds) {
+		clusterPod.Spec.ActiveDeadlineSeconds = desiredPod.Spec.ActiveDeadlineSeconds
 	}
+}
 
-	return unstructuredObj, nil
+func updatePodTolerations(desiredPod, clusterPod *corev1.Pod) {
+	for _, desiredToleration := range desiredPod.Spec.Tolerations {
+		for _, clusterToleration := range clusterPod.Spec.Tolerations {
+			if clusterToleration == desiredToleration {
+				goto skip
```

Comment:
First of all, I don't think it's wrong to use the goto statement. If accepted, this will be the first time the Karmada project has used the goto language, which means more goto languages are likely to follow. In view of the disadvantages of the goto statement, I'm not sure if this operation is reasonable. Ask @RainbowMango to help take a look.
/cc @RainbowMango